funcs/func_groupcount.c

```diff
@@ -237,6 +611,539 @@ static struct ast_custom_function group_function = {
 	.write = group_function_write,
 };
 
+
+/* GROUP_VAR and related */
+
+static int group_var_function_read(struct ast_channel *chan, const char *cmd,
+			       char *data, char *buf, size_t len)
+{
+	char group[MAX_GROUP_LEN] = "";
+	char category[MAX_CATEGORY_LEN] = "";
+	const char *variable_value;
+	AST_DECLARE_APP_ARGS(args,
+		AST_APP_ARG(groupcategory);
+		AST_APP_ARG(varname);
+	);
+	AST_STANDARD_APP_ARGS(args, data);
+
+	buf[0] = '\0';
+
+	if (ast_strlen_zero(args.groupcategory) || ast_strlen_zero(args.varname)) {
+		ast_log(LOG_WARNING, "Syntax GROUP_VAR(group[@category],<varname>)\n");
+		return -1;
+	}
+
+	if (ast_app_group_split_group(args.groupcategory, group, sizeof(group), category, sizeof(category))) {
+		return -1;
+	}
+
+	variable_value = ast_app_group_get_var(group, category, args.varname);
+
+	if (variable_value) {
+		ast_copy_string(buf, variable_value, len);
+	}
+
+	return 0;
+}
+
+static int group_var_function_write(struct ast_channel *chan, const char *cmd,
+				char *data, const char *value)
+{
+	char group[MAX_GROUP_LEN] = "";
+	char category[MAX_CATEGORY_LEN] = "";
+	int result;
+	AST_DECLARE_APP_ARGS(args,
+		AST_APP_ARG(groupcategory);
+		AST_APP_ARG(varname);
+	);
+	AST_STANDARD_APP_ARGS(args, data);
+
+	if (!value) {
+		value = "";
+	}
+
+	if (ast_strlen_zero(args.groupcategory) || ast_strlen_zero(args.varname)) {
+		ast_log(LOG_WARNING, "Syntax GROUP_VAR(group[@category],<varname>)=<value>)\n");
+		return -1;
+	}
+
+	ast_app_group_split_group(args.groupcategory, group, sizeof(group), category, sizeof(category));
+	result = ast_app_group_set_var(chan, group, category, args.varname, value);
+
+	if (result != 0) {
+		/* We know we're not passing any NULL values, so our error has to be 'non-exist' */
+		ast_log(LOG_WARNING, "GROUP_VAR() Variable set failed (group doesn't exist)");
+		return -1;
+	}
+
+	return 0;
+}
+
+static int manager_group_set(struct mansession *s, const struct message *m)
+{
+	const char *channel = astman_get_header(m, "Channel");
+	const char *group = astman_get_header(m, "Group");
+	const char *category = astman_get_header(m, "Category");
+
+	char *group_category;
+	struct ast_channel *chan = NULL;
+
+	if (ast_strlen_zero(channel)) {
+		astman_send_error(s, m, "Channel not specified.");
+		return AMI_SUCCESS;
+	}
+
+	chan = ast_channel_get_by_name(channel);
+	if (!chan) {
+		astman_send_error(s, m, "Channel not found.");
+		return AMI_SUCCESS;
+	}
+
+	if (!group) {
+		group = "";
+	}
+
+	if (!category) {
+		category = "";
+	}
+
+	group_category = ast_malloc(strlen(group) + strlen(category) + 2); /* 1 for @, 1 for NULL*/
```

Comment:
You need to check for malloc failure

Reply:
You're still not checking for malloc failure here.
Either way, don't use ast_malloc, use ast_asprintf.